core/src/test/java/brooklyn/entity/basic/ServiceStateLogicTest.java

```diff
@@ -240,7 +254,33 @@ public void testManuallySettingIndicatorsOnApplications() throws Exception {
         assertAttributeEquals(entity, Attributes.SERVICE_UP, true);
         assertAttributeEquals(entity, Attributes.SERVICE_STATE_ACTUAL, Lifecycle.RUNNING);
     }
-        
+
+    @Test
+    public void testQuorumWithStringStates() {
+        final DynamicCluster cluster = app.createAndManageChild(EntitySpec.create(DynamicCluster.class)
+                .configure(DynamicCluster.MEMBER_SPEC, EntitySpec.create(TestEntityWithoutEnrichers.class))
+                .configure(DynamicCluster.INITIAL_SIZE, 1));
+
+        RecordingSensorEventListener r = new RecordingSensorEventListener();
+        app.subscribe(cluster, Attributes.SERVICE_STATE_ACTUAL, r);
+
+        cluster.start(ImmutableList.of(new SimulatedLocation()));
```

Comment:
There is an `app.newSimulatedLocation()`, which uses the `LocationSpec` and `LocationManager` to instantiate the location.